src/org/zaproxy/zap/view/widgets/WritableFileChooser.java

```diff
@@ -26,6 +28,19 @@ public WritableFileChooser(File currentDirectory) {
 	public void approveSelection() {
 		File selectedFile = getSelectedFile();
 
+		if (!selectedFile.exists()) {
+			selectedFile = selectedFile.getParentFile();
```

Comment:
It should not be saved in this variable, otherwise the following checks would not be correct (e.g. would be checking that the grand-parent directory is writable instead of the parent directory).